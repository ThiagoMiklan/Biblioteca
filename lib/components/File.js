"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _Icon = _interopRequireDefault(require("./Icon"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  icon_name: 'string',
  icon_definition: 'string',
  definition: 'string',
  label: 'string',
  filename: 'string',
  ref: 'string',
  onChange: 'function',
  custom: 'object'
};

var File = function File(props) {
  (0, _type_validations.validate)(props, props_obj, "File");
  var definition = props.definition == undefined ? "file" : "file " + props.definition;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: definition
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "file-label"
  }, /*#__PURE__*/_react.default.createElement("input", (0, _extends2.default)({
    onChange: props.onChange,
    className: "file-input",
    type: "file",
    ref: props.ref
  }, props.custom)), /*#__PURE__*/_react.default.createElement("span", {
    className: "file-cta"
  }, props.icon_name != undefined && /*#__PURE__*/_react.default.createElement(_Icon.default, {
    definition: props.icon_definition,
    icon_name: props.icon_name
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "file-label"
  }, props.label)), definition.includes("has-name") == true && props.filename != undefined && /*#__PURE__*/_react.default.createElement("span", {
    className: "file-name"
  }, props.filename)));
};

var _default = File;
exports.default = _default;