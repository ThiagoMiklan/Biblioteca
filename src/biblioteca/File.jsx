// @flow
import React from 'react'
import PropTypes from 'prop-types';
import Icon from './Icon';



type Props = {
    icon_name?: string,
    icon_definition?: string,
    definition?: string,
    label?: string,
    filename?:string,
    ref?: string,
    onChange?: ()=> void
}

const File = (props:Props)=>{
  var definition = (props.definition == undefined)? "file": "file "+props.definition;
  
    return <div className={definition}>
    <label className="file-label">
      <input onChange={props.onChange} className="file-input" type="file" ref={props.ref}/>
      <span className="file-cta">
        {(props.icon_name != undefined) && <Icon definition={props.icon_definition} icon_name={props.icon_name}/>}
        <span className="file-label">
            {props.label}
        </span>
        </span>
        {definition.includes("has-name") == true && props.filename != undefined &&
          <span className="file-name">
              {props.filename}
          </span>
        }
    </label>
   
  </div>
}

File.propTypes ={
    icon_name: PropTypes.string,
    icon_definition: PropTypes.string,
    definition: PropTypes.string,
    label: PropTypes.string,
    onChange: PropTypes.func
}

export default File;