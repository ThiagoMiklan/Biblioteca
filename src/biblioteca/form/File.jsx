import React from 'react'
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import Icon from '../elements/Icon';

const File = (props)=>{
    return <div class="file">
    <label class="file-label">
      <input class="file-input" type="file" name="resume"/>
        <Icon definition={props.icon_definition} icon_name={props.icon_name}/>
        <span class="file-label">
            {props.label}
        </span>
      
    </label>
  </div>
}

File.propTypes ={
    icon_name: PropTypes.string,
    icon_definition: PropTypes.string,
    definition: PropTypes.string,
    label: PropTypes.string
}

export default File;