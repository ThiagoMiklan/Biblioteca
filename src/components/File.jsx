// @flow
import React from 'react'
import Icon from './Icon';
import { validate } from '../tools/type_validations';


type Props = {
  icon_name?: string,
  icon_definition?: string,
  definition?: string,
  label?: string,
  filename?: string,
  ref?: string,
  onChange?: () => void
}

const props_obj = {
  icon_name: 'string',
  icon_definition: 'string',
  definition: 'string',
  label: 'string',
  filename: 'string',
  ref: 'string',
  onChange: 'function'
}

const File = (props: Props) => {
  validate(props, props_obj, "File");

  var definition = (props.definition == undefined) ? "file" : "file " + props.definition;

  return <div className={definition}>
    <label className="file-label">
      <input onChange={props.onChange} className="file-input" type="file" ref={props.ref} />
      <span className="file-cta">
        {(props.icon_name != undefined) && <Icon definition={props.icon_definition} icon_name={props.icon_name} />}
        <span className="file-label">
          {props.label}
        </span>
      </span>
      {definition.includes("has-name") == true && props.filename != undefined &&
        <span className="file-name">
          {props.filename}
        </span>
      }
    </label>

  </div>
}

export default File;