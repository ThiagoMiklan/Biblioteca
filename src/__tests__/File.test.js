import React from 'react';
import ReactDOM from 'react-dom';
import File from '../biblioteca/File';
import renderer from 'react-test-renderer';

var onChange = (e)=>{
    alert("Escolheu o arquivo!")
}

test('test files',()=>{
    const component = renderer
    .create(<div className="container is-fluid">
    <File onChange={onChange} filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_name="fas fa-upload" />
      <br></br>
    <File
      definition="has-name"
      filename="Screenshot 2020/01/01"
      label="Choose File"
      icon_name="fas fa-upload" />
    <br></br>
    <File
      definition="has-name right"
      filename="Screenshot 2020/01/01"
      label="Choose File"
      icon_name="fas fa-upload" />
    <br></br>
    <File
      definition="has-name fullwidth"
      filename="Screenshot 2020/01/01"
      label="Choose File"
      icon_name="fas fa-upload" />
    <br></br>
    <File 
      definition="boxed"
      filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_name="fas fa-upload" />
    <br></br>
      <File 
      definition="boxed has-name"
      filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_name="fas fa-upload" />
      <br></br>
      <File 
      definition="primary"
      filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_name="fas fa-upload" />
      <br></br>
      <File
      definition="has-name info"
      filename="Screenshot 2020/01/01"
      label="Choose File"
      icon_name="fas fa-upload" />  
      <br></br>
      <File 
      definition="boxed warning"
      filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_name="fas fa-upload" />
      <br></br>
      <File 
      definition="boxed has-name danger"
      filename="Screenshot 2020/01/01"
      label="Choose a File"
      icon_definition="info"
      icon_name="fas fa-upload" />
      <br></br>
  </div>)
        .toJSON();
 
 
    expect(component).toMatchSnapshot();
})