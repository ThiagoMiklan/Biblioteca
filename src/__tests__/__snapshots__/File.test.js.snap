// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test files 1`] = `
<div
  className="container is-fluid"
>
  <div
    className="file"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        onChange={[Function]}
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
    </label>
  </div>
  <br />
  <div
    className="file has-name"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
  <div
    className="file has-name is-right"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
  <div
    className="file has-name is-fullwidth"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
  <div
    className="file is-boxed"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
    </label>
  </div>
  <br />
  <div
    className="file is-boxed has-name"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
  <div
    className="file is-primary"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
    </label>
  </div>
  <br />
  <div
    className="file has-name is-info"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
  <div
    className="file is-boxed is-warning"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
    </label>
  </div>
  <br />
  <div
    className="file is-boxed has-name is-danger"
  >
    <label
      class="file-label"
    >
      <input
        class="file-input"
        type="file"
      />
      <span
        className="file-cta"
      >
        <span />
        <span
          className="icon has-text-info"
        >
          <i
            aria-hidden="true"
            className="fas fa-upload"
          />
        </span>
        <span
          class="file-label"
        >
          Choose a File
        </span>
      </span>
      <span
        className="file-name"
      >
        Screenshot 2020/01/01
      </span>
    </label>
  </div>
  <br />
</div>
`;
